File: fxgl-net/src/main/java/com/almasb/fxgl/net/NetworkComponent.java
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.net;

import com.almasb.fxgl.entity.component.Component;

/**
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
public final class NetworkComponent extends Component {

    private static int uniqueID = 0;

    private int id;

    public NetworkComponent() {
        id = uniqueID++;
    }

    public int getId() {
        return id;
    }

    void setId(int id) {
        this.id = id;
    }

    @Override
    public boolean isComponentInjectionRequired() {
        return false;
    }
}

File: fxgl/src/main/kotlin/com/almasb/fxgl/multiplayer/NetworkComponent.kt
/*
 * FXGL - JavaFX Game Library. The MIT License (MIT).
 * Copyright (c) AlmasB (almaslvl@gmail.com).
 * See LICENSE for details.
 */

package com.almasb.fxgl.multiplayer

import com.almasb.fxgl.entity.component.Component

/**
 *
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
class NetworkComponent : Component() {

    companion object {
        private var uniqueID = 0L
    }

    var id: Long = uniqueID++
        internal set

    override fun isComponentInjectionRequired(): Boolean = false
}